test/jdk/java/awt/EventDispatchThread/InterruptEDTTest.java

```diff
@@ -0,0 +1,107 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Robot;
+import java.awt.Toolkit;
+import java.awt.event.InputEvent;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+
+/*
+ * @test
+ * @key headful
+ * @bug 7184401
+ * @summary Verify that events are not lost when EDT is interrupted.
+ * @modules java.desktop/sun.awt
+ * @run main InterruptEDTTest
+ */
+
+public class InterruptEDTTest {
+
+    private static Frame frame = null;
+    private static volatile Thread edt = null;
+    private static volatile Robot robot = null;
+    private static volatile int xLocation;
+    private static volatile int yLocation;
+    private static volatile int width;
+    private static volatile int height;
+
+    public static void main(String args[]) throws Exception {
+        try {
+            robot = new Robot();
+            EventQueue.invokeAndWait(() -> {
+                edt = Thread.currentThread();
+                frame = new Frame("Frame");
+                frame.addMouseListener(new MouseAdapter() {
+                    public void mouseClicked(MouseEvent me) {
+                        System.out.println("Mouse clicked Event: " + me);
+                    }
+                });
+                frame.setBounds(350, 50, 400, 400);
+                frame.setLocationRelativeTo(null);
+                frame.setVisible(true);
+            });
+            ((sun.awt.SunToolkit) (Toolkit.getDefaultToolkit())).realSync();
+            EventQueue.invokeAndWait(() -> {
+                xLocation = frame.getX();
+                yLocation = frame.getY();
+                width = frame.getWidth();
+                height = frame.getHeight();
+            });
+            ((sun.awt.SunToolkit) (Toolkit.getDefaultToolkit())).realSync();
+            EventQueue.invokeLater(() -> {
+                robot.mouseMove(xLocation + width / 2, yLocation + height / 2);
+                robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+                robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            });
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        edt.interrupt();
+        ((sun.awt.SunToolkit) (Toolkit.getDefaultToolkit())).realSync();
+        try {
+            EventQueue.invokeAndWait(() -> {
```

Comment:
This is duplicate code in Line# 67 to Line# 72.

Reply:
The intent was to ensure latest values are picked up, thereby ensuring the stability of the code.  But I see the point that it does add value since there is not change in the frame position. Will remove it.

Reply:
[Fixed Review Comments: Removed redundant code](https://github.com/openjdk/jdk/pull/10784/commits/ad0a62400b5bcd95c01174e85ee8c7834be39fed)